src/medCore/gui/viewContainers/medViewContainer.cpp

```diff
@@ -829,3 +859,129 @@ void medViewContainer::removeColorIndicator(QColor color)
 {
     d->poolIndicator->removeColorIndicator(color);
 }
+
+/**
+ * @brief saves the scene in a XML file
+ * Saves views (all layers), and toolboxes parameters
+ * 	expected tree is as follow:
+ *  workingDir (user-defined)
+ * 		|-viewID0
+ * 			|-mapping.xml
+ * 			|-layerID01.xml
+ * 			|-layerID02.xml
+ * 			|-data_file (as saved by the dedicated writer)
+ */
+QString medViewContainer::saveScene()
+{
+    QString dirPath = QFileDialog::getExistingDirectory(this, tr("Open Directory"),"/home",QFileDialog::ShowDirsOnly);
+    QDir workingDir(dirPath);
+    QDomDocument doc("xml");
+    QDomElement root=doc.createElement("scene");
+    doc.appendChild(root);
+
+    QString subDirName;
+
+    medAbstractLayeredView* layeredView=dynamic_cast<medAbstractLayeredView*>(view());
+    if (layeredView==0)
+    {
+        qDebug()<< Q_FUNC_INFO <<", dynamic_cast<medAbstractLayeredView*>(view()) failed";
+        medMessageController::instance()->showError("Scene saving failed", 5000);
+        return "";
+    }
+    else
+    {
+        subDirName="view"+QUuid::createUuid ().toString();
+        if(!workingDir.mkdir(subDirName))
+        {
+            QString errorMessage = ", Failed to create new directory: " + subDirName;
+            qDebug()<< Q_FUNC_INFO  << errorMessage;
+            medMessageController::instance()->showError(errorMessage, 5000);
+        }
+        workingDir.cd(subDirName);
+        QString generatedPath=workingDir.canonicalPath()+"/mapping.xml";
+        QDomElement layeredViewInfo=doc.createElement("layeredView");
+        layeredViewInfo.setAttribute("path",subDirName+"/mapping.xml");
+        layeredViewInfo.setAttribute("id",layeredView->identifier());
+        root.appendChild(layeredViewInfo);
+        layeredView->write(generatedPath);
+        workingDir.cdUp();
+    }
+    QString generatedPath=workingDir.canonicalPath()+"/globalMapping.xml";
+
+    QFile file(generatedPath);
+    if (!file.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+    {
+        return "";
+    }
+    else
+    {
+        QTextStream out(&file);
+        out << doc.toString();
+        QString path = dirPath + "/" + subDirName;
+        return path;
+    }
```

Comment:
if you want to be really clear you can:

 ```c++
     medAbstractLayeredView* layeredView=dynamic_cast<medAbstractLayeredView*>(view());
    if (layeredView)
    {
        subDirName="view"+QUuid::createUuid ().toString();
        if(workingDir.mkdir(subDirName))
        {
            workingDir.cd(subDirName);
            QString generatedPath=workingDir.canonicalPath()+"/mapping.xml";
            QDomElement layeredViewInfo=doc.createElement("layeredView");
            layeredViewInfo.setAttribute("path",subDirName+"/mapping.xml");
            layeredViewInfo.setAttribute("id",layeredView->identifier());
            root.appendChild(layeredViewInfo);
            layeredView->write(generatedPath);
            workingDir.cdUp();

            QString generatedGlobalPath=workingDir.canonicalPath()+"/globalMapping.xml";

            QFile file(generatedGlobalPath);
            if (file.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
            {
                QTextStream out(&file);
                out << doc.toString();
                QString path = dirPath + "/" + subDirName;
                return path;
            }
        }
        else
        {
            QString errorMessage = ", Failed to create new directory: " + subDirName;
            qDebug()<< Q_FUNC_INFO  << errorMessage;
            medMessageController::instance()->showError(errorMessage, 5000);
        }
    }
    else
    {
        qDebug()<< Q_FUNC_INFO <<", dynamic_cast<medAbstractLayeredView*>(view()) failed";
        medMessageController::instance()->showError("Scene saving failed", 5000);
    }
    return "";
```
We this kind of construction, we can easily understand when the code stops, continues and when we display a user message :)